### PR TITLE
[Merged by Bors] - Shader Processor: process imported shader

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- I want to be able to use `#ifdef` and other processor directives in an imported shader

## Solution

- Process imported shader strings

### CODE REVIEW

**Review by cart — COMMENTED** [review]
I don't think we actually need to loop here / reprocess the same lines over and over. Once a line has been processed, it should never needs to be re-processed. I think we could make this more efficient by just recursively running the shader processor on import text prior to appending it to the final output.

**Comment by mockersf** [comment]
> I don't think we actually need to loop here

It's to have the possibility of nested imports. It's probably not necessary to loop over everything until nothing changed, but I don't think it should be too long to process text...
I added a test for nested import.

File: crates/bevy_render/src/render_resource/shader.rs [hunk 1]
```diff
@@ -362,46 +362,59 @@ impl ShaderProcessor {
...
-                final_string.push('\n');
             }
-        }
-
-        final_string.pop();
+            processed.pop();
+            if !changed {
+                break processed;
+            }
+            processing = processed.clone();
```
[the remark this answers is not in the document]
Comment (mockersf) [reply crates/bevy_render/src/render_resource/shader.rs]: nop, not needed 👍

**Comment by mockersf** [comment]
no more `loop`...

**Comment by cart** [comment]
Awesome. Thats exactly what I was thinking of :)

**Pull request closed without merging**